tests/Zend/Loader/AutoloaderTest.php

```diff
@@ -340,15 +340,32 @@ public function testAutoloadShouldLoadClassWhenNamespaceIsRegisteredAndClassfile
         $this->assertTrue(class_exists('ZendLoaderAutoloader_Foo', false));
     }
 
-    public function testAutoloadShouldNotSuppressFileNotFoundWarningsWhenFlagIsDisabled()
+    public function testAutoloadShouldNotSuppressParseErrorWhenSuppressNotFoundWarningsFlagIsDisabled()
     {
+        if (version_compare(PHP_VERSION, '7.0.0', '<')) {
+            $this->markTestSkipped(__METHOD__ . ' requires PHP version 7.0.0 or greater');
+        }
+
         $this->addTestIncludePath();
         $this->autoloader->suppressNotFoundWarnings(false);
         $this->autoloader->registerNamespace('ZendLoaderAutoloader');
-        set_error_handler(array($this, 'handleErrors'));
-        $this->assertFalse(Zend_Loader_Autoloader::autoload('ZendLoaderAutoloader_Bar'));
-        restore_error_handler();
-        $this->assertNotNull($this->error);
+        try {
+            $this->assertFalse(Zend_Loader_Autoloader::autoload('ZendLoaderAutoloader_ZFAutoloadParseError'));
+        } catch (ParseError $e) {
+        }
+
+        $this->assertInstanceOf('ParseError', $e);
+    }
+
+    public function testAutoloadShouldSuppressParseErrorWhenSuppressNotFoundWarningsFlagIsEnabled()
+    {
+        if (version_compare(PHP_VERSION, '7.0.0', '<')) {
+            $this->markTestSkipped(__METHOD__ . ' requires PHP version 7.0.0 or greater');
+        }
+        $this->addTestIncludePath();
+        $this->autoloader->suppressNotFoundWarnings(true);
+        $this->autoloader->registerNamespace('ZendLoaderAutoloader');
+        $this->assertFalse(Zend_Loader_Autoloader::autoload('ZendLoaderAutoloader_ZFAutoloadParseError'));
```

Comment:
As part of these changes, I discovered that the Zend_Loader_Autoloader class has a "suppressNotFoundWarnings" flag which will call Zend_Loader with an `@` before it to suppress all errors.

However, that approach also cause other legitimate errors (like parse errors) to be suppressed, which can be difficult to debug. Thus, I decided not to make use of this flag, but went ahead and made some tests covering this behavior. I'm not doing it for <7.0.0 since it's difficult to catch a parse error in that case.